### DISCUSSION TITLE
Shield ignores AuthToken.php config

### DISCUSSION THREAD

**Discussion by alxjzx100** [codeigniter4/shield#1167] [General]
Have a feelling that I missed something, I copied this file (AuthToken.php) from vendor folder to app/Config but codeigniter ignores it. 

I tried to change the $authenticatorHeader but it has no effect. If I decare this variable at Auth.php - everything works.

What I did wrong?

**Comment by datamweb** [comment]
Make sure you set the namespace  correctly. A correct example is as follows.

```php
<?php

 // app/Config/AuthToken 

declare(strict_types=1);

namespace Config;

use CodeIgniter\Shield\Config\AuthToken as ShieldAuthToken;

/**
 * Configuration for Token Auth and HMAC Auth
 */
class AuthToken extends ShieldAuthToken
{
    /**
     * --------------------------------------------------------------------
     * Name of Authenticator Header
     * --------------------------------------------------------------------
     * The name of Header that the Authorization token should be found.
     * According to the specs, this should be `Authorization`, but rare
     * circumstances might need a different header.
     */
    public array $authenticatorHeader = [
        'tokens' => 'Authorization',
        'hmac'   => 'Authorization',
    ];
}
```

**Reply by alxjzx100** [reply]
Thanks for the quick reply, just double-checked - it identical to your example ...

**Comment by alxjzx100** [comment]
I'm a little confused with that code in vendor/codeigniter4/shield/src/Filters/TokenAuth.php


```php
$authenticator = auth('tokens')->getAuthenticator();
$result = $authenticator->attempt([
       'token' => $request->getHeaderLine(setting('Auth.authenticatorHeader')['tokens'] ?? 'Authorization'),
 ]);`
``` 

I'm not sure how setting() works, but if I change 'Auth.authenticatorHeader' to '**AuthToken**.authenticatorHeader' - everything works

**Reply by datamweb** [reply]
Thanks, this is a bug. And your solution is correct.
Also, this issue should be resolved in:
https://github.com/codeigniter4/shield/blob/0aaa04b11edd086694fef8f81f0e4a7bc8437539/src/Filters/HmacAuth.php#L36
If you can send a PR.

**Comment by datamweb** [comment]
Bug fixed by #1169
